src/TConsole.cpp

```diff
@@ -2594,3 +2619,206 @@ void TConsole::slot_reloadMap(QList<QString> profilesList)
+            qDebug() << "TConsole::removeSingleWordFromSet(...) ALERT - failed to get a write lock to access mWordSet_shared and loaded shared hunspell dictionary, retrying...";
+            sharedDictionaryResult = pMudlet->removeSingleWordFromSet(word);
+        }
+
+        if (sharedDictionaryResult.second) {
+            // Successfully added word:
+            result.first = true;
+        } else {
+            // Word already present
+            result.second = errMsg.arg(word);
+        }
+    }
+
+    return result;
+}
+
+void TConsole::setSystemSpellDictionary(const QString& newDict)
+{
+    if (!newDict.isEmpty() && mSpellDic != newDict) {
+        mSpellDic = newDict;
+
+        // This is duplicated (and should be the same as) the code in:
+        // (void) dlgProfilePreferences::initWithHost(Host*)
+        QString path;
+        const QString& currentDictionary = mpHost->getSpellDic();
+#if defined(Q_OS_MACOS)
+        path = QStringLiteral("%1/../Resources/").arg(QCoreApplication::applicationDirPath());
+        mudlet::self()->mUsingMudletDictionaries = true;
+#elif defined(Q_OS_FREEBSD)
+        if (QFile::exists(QStringLiteral("/usr/local/share/hunspell/%1.aff").arg(currentDictionary))) {
+            path = QLatin1String("/usr/local/share/hunspell/");
+            mudlet::self()->mUsingMudletDictionaries = false;
+        } else if (QFile::exists(QStringLiteral("/usr/share/hunspell/%1.aff").arg(currentDictionary))) {
+            path = QLatin1String("/usr/share/hunspell/");
+            mudlet::self()->mUsingMudletDictionaries = false;
+        } else if (QFile::exists(QStringLiteral("%1/../../src/%2.aff").arg(QCoreApplication::applicationDirPath(), currentDictionary))) {
+            // From debug or release subdirectory of a shadow build directory alongside the ./src one:
+            path = QStringLiteral("%1/../../src/").arg(QCoreApplication::applicationDirPath());
+            mudlet::self()->mUsingMudletDictionaries = true;
+        } else if (QFile::exists(QStringLiteral("%1/../src/%2.aff").arg(QCoreApplication::applicationDirPath(), currentDictionary))) {
+            // From shadow build directory alongside the ./src one:
+            path = QStringLiteral("%1/../src/").arg(QCoreApplication::applicationDirPath());
+            mudlet::self()->mUsingMudletDictionaries = true;
+        } else {
+            // From build within ./src
+            path = QStringLiteral("%1/").arg(QCoreApplication::applicationDirPath());
+            mudlet::self()->mUsingMudletDictionaries = true;
+        }
+#elif defined(Q_OS_LINUX)
+        if (QFile::exists(QStringLiteral("/usr/share/hunspell/%1.aff").arg(currentDictionary))) {
+            path = QLatin1String("/usr/share/hunspell/");
+            mudlet::self()->mUsingMudletDictionaries = false;
+        } else if (QFile::exists(QStringLiteral("%1/../../src/%2.aff").arg(QCoreApplication::applicationDirPath(), currentDictionary))) {
```

Comment:
I think it is an OS implementation detail as to where *they* keep their dictionaries and for *Mudlet* it is depending on what sort of build it is - if an installer version we will want to use the ones we bundled (if any, and there not being any already on the system we can use which are more likely to be to the users own requirements).

In reality we are looking around to find the best available ones and they are listed in descending order of likely usefulness (IMHO) given all the different platforms we are working with. Once we find something then we are happy (and that is why I indicate the locations in the individual tooltips - just in case the end user *does* have multiple sets of dictionaries around, like say, if they are working on the source code and want to know what is going on...)

Reply:
Wouldn't it be better to collect the most amount of dictionaries possible?